sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -754,11 +754,16 @@ class Analyzer(
    * a logical plan node's children.
    */
   object ResolveReferences extends Rule[LogicalPlan] {
+
+    private val emptyAttrMap = new AttributeMap[Attribute](Map.empty)
```

Comment:
I'd prefer to do [what I suggested previously](https://github.com/apache/spark/pull/22318#discussion_r214678100) as it would be easier to reuse if it will be needed. In this way, next time we need again an empty AttributeMap we need to create a new one and we'd end up with several of them.

cc @cloud-fan @maropu WDYT?

Reply:
@mgaido91 , I agree with you and happy to do it, just saw your concerns about binary compatibility.
@cloud-fan  @maropu please share your thoughts on this.

Reply:
its ok to me.

Reply:
done